server/src/test/java/org/elasticsearch/gateway/ReplicaShardAllocatorIT.java

```diff
@@ -0,0 +1,158 @@
+                new Object[] { randomIntBetween(10, 100) },
+                new Object[] { randomIntBetween(10, 100) },
+            });
+        }
+        CountDownLatch blockRecovery = new CountDownLatch(1);
+        CountDownLatch recoveryStarted = new CountDownLatch(1);
+        MockTransportService transportServiceOnPrimary
+            = (MockTransportService) internalCluster().getInstance(TransportService.class, nodeWithPrimary);
+        transportServiceOnPrimary.addSendBehavior((connection, requestId, action, request, options) -> {
+            if (PeerRecoveryTargetService.Actions.FILES_INFO.equals(action)) {
+                recoveryStarted.countDown();
+                try {
+                    blockRecovery.await(5, TimeUnit.SECONDS);
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                }
+            }
+            connection.sendRequest(requestId, action, request, options);
+        });
+        try {
+            String newNode = internalCluster().startDataOnlyNode();
+            recoveryStarted.await(5, TimeUnit.SECONDS);
+            // Index more documents and flush to destroy sync_id and remove the retention lease (as file_based_recovery_threshold reached).
+            execute("insert into doc.test (x) values (?)", new Object[][] {
+                new Object[] { randomIntBetween(10, 100) },
+                new Object[] { randomIntBetween(10, 100) },
+            });
+            execute("optimize table doc.test with (flush = true, max_num_segments = 1)");
+            assertBusy(() -> {
+                execute("select unnest(retention_leases['leases']['id']) from sys.shards where table_name = 'test'");
+                for (var row : response.rows()) {
+                    assertThat(row[0], not(equalTo((ReplicationTracker.getPeerRecoveryRetentionLeaseId(discoNodeWithReplica.getId())))));
+                }
+            });
+            // AllocationService only calls GatewayAllocator if there are unassigned shards
+            execute("""
+                create table doc.dummy (x int)
+                with ("routing.allocation.require.attr" = 'not-found')
+            """);
+            internalCluster().startDataOnlyNode(nodeWithReplicaSettings);
+            // need to wait for events to ensure the reroute has happened since we perform it async when a new node joins.
+            client().admin().cluster()
+                .prepareHealth(indexName)
+                .setWaitForYellowStatus()
+                .setWaitForEvents(Priority.LANGUID)
+                .execute()
+                .get(5, TimeUnit.SECONDS);
+            blockRecovery.countDown();
+            ensureGreen(indexName);
+            assertThat(internalCluster().nodesInclude(indexName), hasItem(newNode));
+            //for (RecoveryState recovery : client().admin().indices().prepareRecoveries(indexName).get().shardRecoveryStates().get(indexName)) {
```

Comment:
debug traces? at least the original test does not contain it...

Reply:
Right - it came in later in a patch that we already applied - I commented it out because `prepareRecoveries` doesn't exist in CrateDB and the test currently fails as is before it gets to this line.

Any suggestion how to handle it?  I suspected the failure my be related to other missing patches in the gateway area which is why I didn't spend too much time debugging.

Reply:
I see. Shall we maybe create an issue for this test (which we could also use at the ignore annotation) to track it?

Reply:
As discussed, we'll track that by a project card.